extensions/amp-anim/0.1/amp-anim.js

```diff
@@ -84,6 +82,7 @@ export class AmpAnim extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    observe(this.element, () => this.updateInViewport_());
```

Comment:
I might be paranoid, but this is slightly different than what we had before. To make it equivalent, you'd need to do something like this:

```
layoutCallback() {
  ...
  return this.loadPromise(img).then(() => observe(this.element, ...));
}
```

I.e. this way it's guaranteed that the inViewport will not arrive before loading has completed.

Reply:
Good point. I missed that nuance, since this layoutCallback returns a promise.